sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -298,3 +299,36 @@ case class StringLength(child: Expression) extends UnaryExpression with ExpectsI
 
   override def prettyName: String = "length"
 }
+
+/**
+ * Returns string, with the first letter of each word in uppercase,
+ * all other letters in lowercase. Words are delimited by whitespace.
+ */
+case class InitCap(child: Expression) extends UnaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = StringType
+
+  override def inputTypes: Seq[DataType] = Seq(StringType)
+
+  override def eval(input: InternalRow): Any = {
+    val string = child.eval(input)
+    if (string == null) {
+      null
+    }
+    else if (string.toString.length == 0) {
+      return UTF8String.fromString(string.toString)
+    }
+    else {
+      val str: String = string.toString
+      val sb = new StringBuilder
```

Comment:
It is in Scala sdk. Could you improve it if you have time? just add blank string check  and null string check, and will felicitate many callers of StringBuilder.